File: java/src/LivingThings/Animals/Domba.java
import java.util.concurrent.ThreadLocalRandom;

// nextInt is normally exclusive of the top value,
// so add 1 to make it inclusive

public class Domba extends FarmAnimal implements LivingThings{
   char t_rumput ='@';
   char tempat ='x';

    final int randX =1;
    final int randY =1;

    //private int Full;
    private int posX;
    private int posY;
    private int countlambMeat;

    //Renderable c;

    public Domba (int posX ,int posY,Renderable c){
       
        super(posX,posY);
       // c.setElement(posX,posY)
        //Cell c = new Cell();
        c.setElement(posX,posY,'D');
        this.countlambMeat =0;
        this.posX = posX;
        this.posY = posY;
        //this.Full = FULLMAX;


    }
    public void move (Renderable c ){
        int x = ThreadLocalRandom.current().nextInt(-randX, randY + 1);
        int y =ThreadLocalRandom.current().nextInt(-randX, randY);
        //System.out.println("x : " + x);
        //System.out.println("y : " +y);
        
        if (this.posX+x>=0 && this.posX+x<15&& this.posY+y>=0 && this.posY +y <15 && hidup() && c.getElement(this.posX+x,this.posY+y)== ' '){
            
            if (c.getTypeLand(this.posX+x,this.posY+y)==tempat){
                c.setElement(posX,posY,' ');
                this.posX+=x;
                this.posY +=y;
                c.setElement(this.posX,this.posY,'D');
                setFull(getFull()-1);
                
            } 
            
            //System.out.println(getFull());
            //c.setElement(this.posX,this.posY,'D');
        } else {
            if (!hidup()){
                c.setElement(this.posX,this.posY,' ');
            }
            
        }
    }
    public void talk() {
        System.out.println("Mbekkkk~~~");
    }
    public void eat(Renderable c){
        if (!hidup()){
                //setStatus(false);
                c.setElement(this.posX,this.posY,' ');
        } else {
            if(c.getRumput(this.posX,this.posY)==t_rumput){
                this.countlambMeat =MAX;
                setFull(FULLMAX);
                c.kosongRumput(this.posX,this.posY);
         } else move(c);
        
        }
       
    }
    public String getChar() {
        return "A";
    }

    public void getProduct() {
        this.countlambMeat =0;
    }
    public Integer getCountProduct() {
        return countlambMeat;
    }
    /*public Integer getFull() {
        return Full;
    }*/

    /*public void setFull(int Full){
        this.Full = Full;
    }*/
    public void print() {
        System.out.println("domba " + countlambMeat + " Full : "+ getFull()+"pos :  "+ posX +" "+ posY);
    }
}

File: src/domba.cpp
#include <iostream>
#include <string>
#include "FarmAnimal.h"

#include "Meatproducing.h"
#include "domba.h"

using namespace std;

const int randX =3;
const int randY =3;

const char t_rumput ='@';
const char tempat ='x';

domba::domba(int posX, int posY):FarmAnimal(posX, posY), Meatproducing(posX, posY) {
    Full = FULLMAX;
    countlambMeat = 0;
    this->posX = FarmAnimal::getPosX();
    this->posY = FarmAnimal::getPosY();
    status = true;
    //_c.setElement(posX,posY,'S') ;
}

domba::~domba() {
    //_c.setElement(posX,posY,' ')
    this->posX = 0;
    this->posY = 0;
    Full = 0;
    countlambMeat = 0;
   // cout << "One of your lambs has died" << endl;
}

void domba::move(Cell&_c) {
    //int xa = FarmAnimal::getPosX();
    srand(time(NULL));
    int x = rand()%(randX) - 1;
    int y = rand()%(randY) - 1;
    //_c.setElement(posX,posY,' ');
    cout<<" x , y "<<x<< " " <<y<<endl;
     if(posX+x>=0 && posX+x<15 && posY+y>=0 && posY+y<15&&status){
             if(_c.getTypeLand(this->posX+x,this->posY+y)==tempat)
                {
                    this->posX+=x;
                    this->posY+=y;
                }
    
     }

    //FarmAnimal::setPosX(xa);
    _c.setElement(posX,posY,'S');
    //cout<<"pos MOVE : "<< this->posX<<" " << this->posY<<endl;
    
    Full--;
}

void domba::talk() {
    cout << "Mbekk~~" << endl;
}

void domba::eat(Cell& _c) {
    if((_c.getRumput(this->posX,this->posY)== t_rumput)){
        Full = FULLMAX;
        countlambMeat++;
        if (countlambMeat >MAX) {
        countlambMeat = MAX;
        }
        _c.kosongRumput(this->posX,this->posY);
    } else {
        move(_c);
        if (Full <= dead){
            status =false;
            _c.setElement(posX,posY,' ');
            cout<<"One of your lambs has died"<<endl;
        }        
    }
}

string domba::getChar() {
    return "D";
}

string domba::getProduct() {
    return "";
}

//void setProduct(string lambMeat);
int domba::getcountlambMeat() const {
    return countlambMeat;
}

int domba::getFull() {
    return Full;
}

void domba::setFull(int Full) {
    this->Full=Full;
}

void domba::Print() {
    cout << "LM ada " << countlambMeat << endl;
    cout << "Full : " <<  Full << endl;
    cout << "pos : " <<  this->posX << " " << this->posY << endl;
}
